cmd/commands/service/command.go

```diff
@@ -66,24 +65,25 @@ func NewCommand(licenseCommandName string) *cli.Command {
 				os.Exit(2)
 			}
 
+			errorChannel := make(chan error, 3)
 			nodeOptions := cmd.ParseFlagsNode(ctx)
 			if err := di.Bootstrap(nodeOptions); err != nil {
 				return err
 			}
-			if err := di.BootstrapServices(nodeOptions); err != nil {
-				return err
-			}
+			go func() { errorChannel <- di.Node.Wait() }()
+
+			cmd.RegisterSignalCallback(func() { errorChannel <- nil })
 
 			cmdService := &serviceCommand{
-				identityHandler: identity_selector.NewHandler(
-					di.IdentityManager,
-					di.MysteriumAPI,
-					identity.NewIdentityCache(nodeOptions.Directories.Keystore, "remember.json"),
-					di.SignerFactory,
-				),
-				di: &di,
+				tequilapi: client.NewClient(nodeOptions.TequilapiAddress, nodeOptions.TequilapiPort),
+				runErrors: errorChannel,
 			}
-			return cmdService.Run(ctx)
+
+			go func() {
+				errorChannel <- cmdService.Run(ctx)
+			}()
+
+			return <-errorChannel
```

Comment:
We have 3 components that can return an error, but we are reading only the first one, others became blocked to send error waiting for reader forever. 
With a buffered channel we are receiving all errors not blocking writers.

Reply:
But as you noted above, if we get an error in any of sub components, we instantly return with `return <-errorChannel`. Other sub components are in routines and probably is does not matter if they will block. Execution will be finished anyways.

Reply:
hm, right. Initially, it was added a long time ago because it was not able to stop properly. I'll test different cases with non-buffered channels to make sure how it works.

Reply:
Changed it to unbuffered channel.